lib/rules/prefer-singular-enums.js

```diff
@@ -0,0 +1,39 @@
+const pluralize = require('pluralize');
+
+module.exports = {
+  meta: {
+    docs: {
+      description: 'Prefer singular TypeScript enums.',
+      category: 'Stylistic Issues',
+      recommended: false,
+      uri:
+        'https://github.com/Shopify/eslint-plugin-shopify/blob/master/docs/rules/prefer-singular-enums.md',
+    },
+    fixable: null,
+  },
+  create(context) {
+    function report(node) {
+      const {name} = node;
+
+      context.report({
+        node,
+        message: `Enum '{{name}}' should be singular.`,
+        data: {name},
+      });
+    }
+
+    return {
+      Identifier(node) {
```

Comment:
I would target the enum declarations and check/ report the identifier, rather than the reverse (this is doing a lot of unnecessary checks given how many identifiers there are)